### ISSUE TITLE
isIPv4String Function validating strings that are not proper IPv4 strings

### ISSUE THREAD

**Issue by MclellandG** [ClickHouse/ClickHouse#65213]
The isIPv4String function is returning true for values that are not valid IPv4 strings.   This causes problems downstream when attempting to further process/store this value as IPv4. 

**How to reproduce**
```
select isIPv4String('1.1.1.1\0000')
select isIPv4String('1.1.1.1\0000s4')

┌─isIPv4String('1.1.1.1\0000')─┐
│                            1 │
└──────────────────────────────┘
```
Of course, trying to use this value in other IP functions fails
```
select isIPAddressInRange('1.1.1.1\0000', '1.1.1.1/8')

 isIPAddressInRange DB::Exception: Neither IPv4 nor IPv6 address: '1.1.1.1000'
```
**Expected behavior**
This should return false as it is not a valid IPv4String.

**Comment by Algunenano** [comment]
The problem is in DB::parseIPv4whole, which checks that end is NULL and that's completely incorrect. It should check against `dst` to verify that the full string was parsed.

**Comment by maynk27** [comment]
Can you please assign it to me?

**Comment by alexey-milovidov** [comment]
@maynk27 Thank you, your help will be very appreciated!

**Comment by ChenMiaoi** [comment]
Hello, is this issue still in progress? If not, can you assign it to me?

**Comment by ChenMiaoi** [comment]
**I can't recurring this issue, my current commit is f7c03bfa3ce(master)**

![Image](https://github.com/user-attachments/assets/d0ab4724-022f-44d2-880c-a0dfabc146d7)

Did it have been fixed? or How can I recurring? It seems works well.

**Comment by alexey-milovidov** [comment]
Yes, I fixed it: https://github.com/ClickHouse/ClickHouse/pull/73946

**Comment by ChenMiaoi** [comment]
@alexey-milovidov I just came into contact with clickhouse. if ok, can you recommend some issues that I can do?

**Comment by alexey-milovidov** [comment]
I recommend this: https://github.com/ClickHouse/ClickHouse/issues/54485
